vrpn_Laputa.C

```diff
@@ -0,0 +1,192 @@
+/** @file	vrpn_Laputa.C
+@brief	Drivers for Laputa VR devices.
+
+  @date 2016
+  @copyright 2016 Sensics Inc.
+  @author Sensics Inc, georgiy@sensics.com
+  @license Standard VRPN license.
+*/
+
+// Based on the vrpn_Oculus driver
+
+#include "vrpn_Laputa.h"
+
+#include "vrpn_BaseClass.h" // for ::vrpn_TEXT_NORMAL, etc
+
+VRPN_SUPPRESS_EMPTY_OBJECT_WARNING()
+
+#if defined(VRPN_USE_HID)
+
+// USB vendor and product IDs for the models we support
+static const vrpn_uint16 LAPUTA_VENDOR = 0x2633;
+static const vrpn_uint16 LAPUTA_PRODUCT = 0x0006;
+static const vrpn_int32 NUM_CHANNELS = 11;
+
+vrpn_Laputa::vrpn_Laputa(const char *name, vrpn_Connection *c)
+    : vrpn_Analog(name, c)
+    , vrpn_HidInterface(
+          m_filter = new vrpn_HidProductAcceptor(LAPUTA_VENDOR, LAPUTA_PRODUCT))
+{
+
+    vrpn_Analog::num_channel = NUM_CHANNELS;
+
+    memset(channel, 0, sizeof(channel));
+    memset(last, 0, sizeof(last));
+}
+
+vrpn_Laputa::~vrpn_Laputa() { delete m_acceptor; }
+
+inline void unpackVector(const vrpn_uint8 raw[8], int vector[3])
```

Comment:
This should be at least partially redundant with the `vrpn_unbuffer` functions (the ones from big endian), if I'm understanding the code right. This sort of union funkiness is varying degrees of not defined in C++ depending on what version of the standard you're using.